### PR TITLE
Fixes to 10 coverity defect reports

### PR DESCRIPTION
**Pull request by harveyj1953**

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build please.

**Comment by pcanal** [comment]
Could you resolve the conflicts?

File: hist/hist/src/TH1Merger.cxx [hunk 1]
```diff
@@ -383,16 +383,18 @@ Bool_t TH1Merger::BufferMerge() {
                fH0->Fill(hist->fBuffer[2*i + 2], hist->fBuffer[2*i + 1]);
          }
          if (fH0->fDimension == 2) {
-            auto h2 = dynamic_cast<TH2*>(fH0);
-            R__ASSERT(h2); 
-            for (Int_t i = 0; i < nbentries; i++)
-               h2->Fill(hist->fBuffer[3*i + 2], hist->fBuffer[3*i + 3],hist->fBuffer[3*i + 1] );
+            if (auto h2 = dynamic_cast<TH2*>(fH0)){
+                  R__ASSERT(h2);
```
Comment (vgvassilev) [inline hist/hist/src/TH1Merger.cxx]: This seems like a false positive in coverity, i.e. if h2 is nullptr we will issue a fatal error and abort. The assert in the if-stmt here becomes dead code.

File: hist/hist/src/TH1Merger.cxx [hunk 2]
```diff
@@ -383,16 +383,18 @@ Bool_t TH1Merger::BufferMerge() {
...
+                     h2->Fill(hist->fBuffer[3*i + 2], hist->fBuffer[3*i + 3],hist->fBuffer[3*i + 1] );
+            }
          }
          if (fH0->fDimension == 3) {
-            auto h3 = dynamic_cast<TH3*>(fH0);
-            R__ASSERT(h3); 
-            for (Int_t i = 0; i < nbentries; i++)
-               h3->Fill(hist->fBuffer[4*i + 2], hist->fBuffer[4*i + 3],hist->fBuffer[4*i + 4], hist->fBuffer[4*i + 1] );
+            if (auto h3 = dynamic_cast<TH3*>(fH0)){
+                  R__ASSERT(h3);
```
Comment (vgvassilev) [inline hist/hist/src/TH1Merger.cxx]: Likewise.

File: hist/hist/src/TH1Merger.cxx (hunk 1 above)
Comment (peremato) [inline hist/hist/src/TH1Merger.cxx]: Probably coverity does not know that you will never comeback from  R__ASSERT with a nullptr.

Comment (pcanal) [inline hist/hist/src/TH1Merger.cxx]: This is indeed very likely.  Is there a way to teach coverity about the semantic of that macro?

Comment (Axel-Naumann) [inline hist/hist/src/TH1Merger.cxx]: Coverity has no way of knwing that `::Fatal()` never returns. We can tell it about `::Fatal()`, though. See https://coverity.cern.ch/docs/en/cov_checker_ref.html#killpath

I'll take care of it...

Comment (Axel-Naumann) [inline hist/hist/src/TH1Merger.cxx]: Told Coverity about `Fatal()` in 7c8286d9f6d72cb14b3f7b8a2ce64a2ce50fa8a0 - that should get rid if this "defect".

**Comment by peremato** [comment]
@phsft-bot build

**Comment by peremato** [comment]
@phsft-bot build

**Comment by peremato** [comment]
One failure of the system in one cell.

File: math/mathmore/test/testInterpolation.cxx [hunk 3]
```diff
@@ -29,10 +29,14 @@ void interpolate( const  ROOT::Math::Interpolator & itp, bool drawSame = false )
...
    std::cout << "x[i]     y[i]     deriv[i]     deriv2[i]    integral[i] \n" << std::endl;
    // print result of interpolation
-   const Int_t n = 51;
+   const Int_t n = 50;  //JH replacing n = 51;
    Int_t i = 0;
+   double xi = 0.;
    Float_t xcoord[n], ycoord[n];
-   for ( double xi = 0; xi < 10.01; xi += 0.2) {
+
+   for ( i = 0; i < 50; ++i) {
```
Comment (vgvassilev) [inline math/mathmore/test/testInterpolation.cxx]: Hm, was it intentional to increment `i` twice: once in the loop and once in the loop's body (see last line).

Comment (lmoneta) [inline math/mathmore/test/testInterpolation.cxx]: I don't understand these changes too. The old code looked perfectly fine to me

**Pull request merged**